pandas/core/indexes/datetimelike.py

```diff
@@ -733,6 +740,12 @@ def __add__(self, other):
             elif is_integer_dtype(other) and self.freq is None:
                 # GH#19123
                 raise NullFrequencyError("Cannot shift with no freq")
+            elif is_float_dtype(other):
+                # Explicitly catch invalid dtypes
+                raise TypeError("cannot add {dtype}-dtype to {cls}"
+                                .format(dtype=other.dtype,
+                                        cls=type(self).__name__))
+
             else:  # pragma: no cover
```

Comment:
what hits this then? You want to make this NOT propagate on the reversed ops yes?

Reply:
Many cases still hit this (objectdtype, categoricaldtype, miscellaneous scalars...) catching floatdype is the main obvious one.  Handling object dtype correctly is the next PR.